components/base/kinematicbase/kinematics.go

```diff
@@ -202,3 +222,42 @@ func CollisionGeometry(cfg *referenceframe.LinkConfig) ([]spatialmath.Geometry,
 	}
 	return []spatialmath.Geometry{sphere}, nil
 }
+
+// newValidRegionCapsule returns a capsule which defines the valid regions for a base to be when moving to a waypoint.
+// The valid region is all points that are deviationThreshold (mm) distance away from the line segment between the
+// starting and ending waypoints. This capsule is used to detect whether a base leaves this region and has thus deviated
+// too far from its path.
+func newValidRegionCapsule(starting, desired []referenceframe.Input, deviationThreshold float64) (spatialmath.Geometry, error) {
+	pt := r3.Vector{X: (desired[0].Value + starting[0].Value) / 2, Y: (desired[1].Value + starting[1].Value) / 2}
+
+	capsule, err := spatialmath.NewCapsule(
+		spatialmath.NewZeroPose(),
+		deviationThreshold,
+		2*deviationThreshold+math.Sqrt(math.Pow(desired[0].Value-starting[0].Value, 2)+math.Pow(desired[1].Value-starting[1].Value, 2)),
```

Comment:
[opt] this is also just the first returned argument of `errorState`, maybe a good simplification to reuse this even if it is a little inefficient to contain the math a little.  Could go either way on it though

Reply:
I agree it looks cleaner too, will just use errorState